solidity/test/staker-rewards/MerkleDistributorTest.js

```diff
@@ -0,0 +1,210 @@
+  let keepToken
+  let rewardsDistributor
+
+  before(async () => {
+    keepToken = await KeepToken.new()
+    rewardsDistributor = await ECDSARewardsDistributor.new(keepToken.address)
+  })
+
+  beforeEach(async () => {
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("allocating rewards", () => {
+    it("should successfuly allocate KEEP tokens", async () => {
+      let balance = await keepToken.balanceOf(rewardsDistributor.address)
+      expect(balance).to.eq.BN(0)
+
+      await allocateTokens(
+        testValues.merkleObject0.merkleRoot,
+        testValues.merkleObject0.amountToAllocate
+      )
+
+      balance = await keepToken.balanceOf(rewardsDistributor.address)
+      expect(balance).to.eq.BN(testValues.merkleObject0.amountToAllocate)
+    })
+
+    it("should fail allocating KEEP tokens without prior approval", async () => {
+      await expectRevert(
+        rewardsDistributor.allocate(
+          testValues.merkleObject0.merkleRoot,
+          testValues.merkleObject0.amountToAllocate
+        ),
+        "SafeERC20: low-level call failed"
+      )
+    })
+  })
+
+  describe("claiming rewards", () => {
+    const merkle0 = testValues.merkleObject0
+    const merkle1 = testValues.merkleObject1
+    beforeEach(async () => {
+      await allocateTokens(
+        testValues.merkleObject0.merkleRoot,
+        testValues.merkleObject0.amountToAllocate
+      )
+
+      await allocateTokens(
+        testValues.merkleObject1.merkleRoot,
+        testValues.merkleObject1.amountToAllocate
+      )
+    })
+
+    it("should successfuly claim rewards and emit an event", async () => {
+      const merkleRoot = merkle0.merkleRoot
+      const index = merkle0.claims[0].index
+      const account = merkle0.claims[0].account
+      const amount = merkle0.claims[0].amount
+      const proof = merkle0.claims[0].proof
+
+      const claimed = await rewardsDistributor.claim(
+        merkleRoot,
+        index,
+        account,
+        amount,
+        proof
+      )
+
+      expectEvent(claimed, "RewardsClaimed", {
+        merkleRoot,
+        index,
+        account,
+        amount,
+      })
+    })
+
+    it("should successfuly claim rewards and updated contract balance", async () => {
+      const initialBalance = await keepToken.balanceOf(
+        rewardsDistributor.address
+      )
+
+      let claimedAmounts = new BN(0)
+      for (let i = 0; i < merkle0.claims.length; i++) {
+        const merkleRoot = merkle0.merkleRoot
+        const index = merkle0.claims[i].index
+        const account = merkle0.claims[i].account
+        const amount = merkle0.claims[i].amount
+        const proof = merkle0.claims[i].proof
+
+        const value = new BN(amount)
+        claimedAmounts = claimedAmounts.add(value)
+
+        await rewardsDistributor.claim(
+          merkleRoot,
+          index,
+          account,
+          amount,
+          proof
+        )
+      }
+
```

Comment:
Should we `expect(await keepToken.balanceOf(account)).to.eq.BN(amount)` and do the same after the second claim?